### ISSUE TITLE
disallowQuotedKeysInObjects: should not throw error if at least one of the field should have quotes

### ISSUE THREAD

**Issue by iamstarkov** [jscs-dev/node-jscs#1743]
```
disallowQuotedKeysInObjects: Extra quotes for key at ./src/Grid.jsx :
     7 |  render() {
     8 |    const classes = classNames({
     9 |      'grid': !this.props.fluid,
--------------^
    10 |      'grid--fluid': this.props.fluid,
    11 |    }, this.props.className);
```

lets discus that quotes on every field should not throw an error if at least one of the field should have quotes

**Comment by iamstarkov** [comment]
it makes sense

**Comment by jfelchner** [comment]
FWIW, this functionality is similar to ESLint's [`consistent-as-needed`](http://eslint.org/docs/rules/quote-props.html#consistent-as-needed) option.

**Comment by hzoo** [comment]
I'm putting an [`orphaned`](https://github.com/jscs-dev/node-jscs/issues?q=label%3Aorphaned+is%3Aclosed) label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in core). We can all check back on this list for future rule ideas/options.

We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
